compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -582,13 +582,18 @@ fn incremental_verify_ich<CTX, K, V: Debug>(
 
     let old_hash = tcx.dep_graph().prev_fingerprint_of(dep_node);
 
-    assert_eq!(
-        Some(new_hash),
-        old_hash,
-        "found unstable fingerprints for {:?}: {:?}",
-        dep_node,
-        result
-    );
+    if Some(new_hash) != old_hash {
+        let run_cmd = if let Some(crate_name) = &tcx.sess().opts.crate_name {
+            format!("`cargo clean -p {}` or `cargo clean`", crate_name)
+        } else {
+            "`cargo clean`".to_string()
+        };
```

Comment:
You should always have the crate name available at this point: 
```suggestion
        let run_cmd = format!("`cargo clean -p {}` or `cargo clean`", tcx.crate_name());
```

Reply:
Unfortunately, `tcx` isn't actually a `TyCtct`, so I can't call `crate_name()`

Reply:
Could you move that function to the CTX trait maybe?